src/internal_rules.jl

```diff
@@ -447,6 +447,108 @@ function EnzymeRules.reverse(config, func::Const{typeof(\)}, ::Type{RT}, cache,
     return (nothing,nothing)
 end
 
+const EnzymeTriangulars = Union{
+    UpperTriangular,
+    LowerTriangular,
+    UnitUpperTriangular,
+    UnitLowerTriangular
+}
+
+function EnzymeRules.augmented_primal(
+    config,
+    func::Const{typeof(\)},
+    ::Type{RT},
+    A::Annotation{AT},
+    b::Annotation{BT}
+) where {RT, AT <: EnzymeTriangulars, BT <: Array}
+    cache_A = EnzymeRules.overwritten(config)[2] ? copy(A.val) : A.val
+    cache_A = compute_lu_cache(cache_A, b.val)
+    res = (cache_A \ b.val)::eltype(RT)
+    dres = if EnzymeRules.width(config) == 1
+        zero(res)
+    else
+        ntuple(Val(EnzymeRules.width(config))) do i
+            Base.@_inline_meta
+            zero(res)
+        end
+    end
+    retres = EnzymeRules.needs_primal(config) ? res : nothing
+    cache_res = EnzymeRules.needs_primal(config) ? copy(res) : res
+    cache_b = EnzymeRules.overwritten(config)[3] ? copy(b.val) : nothing
+    cache = NamedTuple{
+        (Symbol("1"), Symbol("2"), Symbol("3"), Symbol("4")),
+        Tuple{typeof(res), typeof(dres), typeof(cache_A), typeof(cache_b)}
+    }((cache_res, dres, cache_A, cache_b))
+    return EnzymeRules.AugmentedReturn{typeof(retres), typeof(dres), Any}(retres, dres, cache)
+end
+
+function EnzymeRules.reverse(
+    config,
+    func::Const{typeof(\)},
+    ::Type{RT},
+    cache,
+    A::Annotation{AT},
+    b::Annotation{BT}
+) where {RT, AT <: EnzymeTriangulars, BT <: Array}
+    y, dys, cache_A, cache_b = cache
+
+    if !EnzymeRules.overwritten(config)[3]
+        cache_b = b.val
+    end
+
+    if EnzymeRules.width(config) == 1
+        dys = (dys,)
+    end
+
+    dAs = if EnzymeRules.width(config) == 1
+        typeof(A) <: Const ? (nothing,) : (A.dval,)
+    else
+        if typeof(A) <: Const
+            ntuple(Val(EnzymeRules.width(config))) do i
+                Base.@_inline_meta
+                nothing
+            end
+        else
+            A.dval
+        end
+    end
+
+    dbs = if EnzymeRules.width(config) == 1
+        if typeof(b) <: Const
+            (nothing,)
+        else
+            (b.dval,)
+        end
+    else
+        if typeof(b) <: Const
+            ntuple(Val(EnzymeRules.width(config))) do i
+                Base.@_inline_meta
+                nothing
+            end
+        else
+            b.dval
+        end
+    end
+
+    for (dA, db, dy) in zip(dAs, dbs, dys)
+        z = transpose(cache_A) \ dy
```

Comment:
Pretty sure this needs to be for the complex case:
```suggestion
        z = adjoint(cache_A) \ dy
```